dev/MRTCore/mrt/Core/src/MRM.cpp

```diff
@@ -899,26 +906,53 @@ STDAPI MrmGetFilePathFromName(_In_ PCWSTR filename, _Outptr_ PWSTR* filePath)
 
     pointerToPath = pathAllocated ? pathAllocated.get() : path;
 
-    size_t lengthInSizeT;
-    RETURN_IF_FAILED(StringCchLengthW(pointerToPath, STRSAFE_MAX_CCH, &lengthInSizeT));
-    length = static_cast<DWORD>(lengthInSizeT);
+    std::unique_ptr<wchar_t, decltype(&MrmFreeResource)> finalPath(nullptr, MrmFreeResource);
 
-    RETURN_IF_FAILED(StringCchLengthW(filename, STRSAFE_MAX_CCH, &lengthInSizeT));
-    DWORD lengthOfName = static_cast<DWORD>(lengthInSizeT);
+    // Will build the path at most twice.
+    // First time using current path. If not exist, do another time with parent path.
+    for (int i = 0; i < 2; i++)
+    {
+        size_t lengthInSizeT;
+        RETURN_IF_FAILED(StringCchLengthW(pointerToPath, STRSAFE_MAX_CCH, &lengthInSizeT));
+        length = static_cast<DWORD>(lengthInSizeT);
+
+        RETURN_IF_FAILED(StringCchLengthW(filename, STRSAFE_MAX_CCH, &lengthInSizeT));
+        DWORD lengthOfName = static_cast<DWORD>(lengthInSizeT);
+
+        RETURN_IF_FAILED(DWordAdd(length, lengthOfName, &length));
+        RETURN_IF_FAILED(DWordAdd(length, 1, &length));
+        RETURN_IF_FAILED(DWordMult(length, sizeof(wchar_t), &size));
+
+        PWSTR rawOutputPath = reinterpret_cast<PWSTR>(MrmAllocateBuffer(size));
+        RETURN_IF_NULL_ALLOC(rawOutputPath);
 
-    RETURN_IF_FAILED(DWordAdd(length, lengthOfName, &length));
-    RETURN_IF_FAILED(DWordAdd(length, 1, &length));
-    RETURN_IF_FAILED(DWordMult(length, sizeof(wchar_t), &size));
+        std::unique_ptr<wchar_t, decltype(&MrmFreeResource)> outputPath(rawOutputPath, MrmFreeResource);
+
+        RETURN_IF_FAILED(StringCchCopyW(outputPath.get(), length, pointerToPath));
+        RETURN_IF_FAILED(StringCchCatW(outputPath.get(), length, filename));
+
+        if (GetFileAttributes(outputPath.get()) != INVALID_FILE_ATTRIBUTES)
```

Comment:
This will also succeed if a directory, not a file, with the same name exists. Might want to explicitly test for and exclude items with `FILE_ATTRIBUTE_DIRECTORY`.

Reply:
Yes. There is little time before release. Also this is no throwing code, and this doesn't use std. 
I was debating whether exclude directory or make code simple. The chance you have  a folder called resources.pri is extremely slim. I can make that check if you insist.

Reply:
Ha! I'm an external so can't insist anything. Agree the risk is low, we can create an issue to track it and move forward.